runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/FileWatcherTask.java

```diff
@@ -0,0 +1,147 @@
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+
+public class FileWatcherTask extends WatcherTask
+{
+    private final Map<Path, byte[]> configHashes;
+    private WatchService watchService;
+
+    public FileWatcherTask(BiConsumer<URL, String> configChangeListener)
+    {
+        super(configChangeListener);
+        this.configHashes = new HashMap<>();
+        try
+        {
+            watchService = FileSystems.getDefault().newWatchService();
+        }
+        catch (IOException ex)
+        {
+            rethrowUnchecked(ex);
+        }
+
+    }
+
+    @Override
+    public boolean run()
+    {
+        try
+        {
+            while (true)
+            {
+                try
+                {
+                    final WatchKey key = watchService.take();
+
+                    for (WatchEvent<?> event : key.pollEvents())
+                    {
+                        final Path changed = ((Path) key.watchable()).resolve((Path) event.context());
+                        if (configHashes.containsKey(changed))
+                        {
+                            String configText = readConfigText(configURLs.get(changed));
+                            byte[] newConfigHash = computeHash(configText);
+                            if (!Arrays.equals(configHashes.get(changed), newConfigHash))
+                            {
+                                configHashes.put(changed, newConfigHash);
+                                configChangeListener.accept(configURLs.get(changed), configText);
+                            }
+                        }
+                    }
+                    key.reset();
+                }
+                catch (InterruptedException ex)
+                {
+                    watchService.close();
+                    break;
+                }
+            }
+        }
+        catch (IOException ex)
+        {
+            rethrowUnchecked(ex);
+        }
+        return true;
+    }
+
+    @Override
+    protected void doInitialConfiguration(URL configURL)
+    {
+        Path configPath = Paths.get(new File(configURL.getPath()).getAbsolutePath());
+        configURLs.put(configPath, configURL);
+        String configText = readConfigText(configURL);
+        configHashes.put(configPath, computeHash(configText));
+
+        try
+        {
+            configPath.getParent().register(watchService, ENTRY_MODIFY, ENTRY_CREATE, ENTRY_DELETE);
+        }
+        catch (IOException ignored)
+        {
+        }
+
+        configChangeListener.accept(configURL, configText);
+    }
+
+    private String readConfigText(URL configURL)
+    {
+        String configText;
+        if (configURL == null)
+        {
+            configText = CONFIG_TEXT_DEFAULT;
+        }
```

Comment:
I don't think we want to be adding semantics about config defaulting into the file watcher, that seems more appropriate at the engine level or similar, so that file vs http watcher doesn't have to replicate defaulting logic.

The watcher should really be dealing with both `null` (not found) and non-`null` values for config.

Reply:
If we get IOException during read, we now let the RegisterTask to detect that and set the the config to default. The watcherTask still registers the WatchService for the file